Doc/reference/datamodel.rst

```diff
@@ -1962,6 +1960,55 @@ Notes on using *__slots__*
   of the iterator's values. However, the *__slots__* attribute will be an empty
   iterator.
 
+Per-attribute docstrings with *__slots__*
```

Comment:
This is too extensive for the reference manual.  I would omit this section entirely.

Instead, focus on correcting lines 1947-1949, perhaps like this, "If a mapping is used, the mapping keys will be used as the slot names, and the *inspect* module will treat the mapping values as attribute docstrings which will be shown in *help()*.